addons/website/static/src/js/content/adapt_content.js

```diff
@@ -16,13 +16,16 @@ document.addEventListener('DOMContentLoaded', () => {
             }
         });
     }
-    // Hack: we move the '#o_search_modal' from the '#header' to
-    // '#o_shared_blocks'. Without this change, when the header has a
-    // 'transform: translate' (when it's fixed), the modal, which is positioned
-    // absolutely, takes the dimensions of the header instead of those of the
-    // 'body'.
+    // To have search modal for all pages, We move '#o_search_modal' from the
+    // '#header' to the shared element '#o_search_block'.
     const searchModalEl = document.querySelector("header#top .modal#o_search_modal");
     if (searchModalEl) {
-        document.querySelector("#o_shared_blocks").appendChild(searchModalEl);
+        const mainEl = document.querySelector("main");
+
+        const searchDiv = document.createElement('div');
+        searchDiv.id = "o_search_block";
```

Comment:
I prefer `o_search_modal_block`. What do you think ?

Reply:
That can be a good name.